### PR TITLE
Products route

### LINKED ISSUE #63: Complete Routes
# User Story
As a **programmer**,
I want to **connect my frontend to mongoDB**,
so I can **stable connection between front- & backend**.

# Description

#Material

# Done

- [x] Connect product routes
- [x] Connect supermarket routes
- [x] ...

### PR DESCRIPTION
**Pull request by LebkuchenCJ**
Fix #63 

## Comments: 
 - This PR will partly fix my issue as I cannot compare it with my supermarkets yet.
 - Products can be added (suggested or your own)
 - You cannot add the same products from the suggestion twice

image

### CODE REVIEW

File: client/src/pages/List.js [hunk 1]
```diff
@@ -24,24 +26,34 @@ function List({ onGroceryListChange }) {
...
-    await postProduct(data);
-    await refetch();
-    setQuery("");
+    const matchedProduct = list.products.find(
+      ({ productId }) => productId === product.id
+    );
+    if (!matchedProduct) {
+      setDisplay(!display);
+      setInputField(false);
+      const newId = Math.floor(Math.random() * (500 - 1)) + 1;
```
Comment (manuelpapa) [inline client/src/pages/List.js:35]: Now that's clever!

File: lib/models/list.js [hunk 2]
```diff
@@ -19,6 +29,23 @@ async function readList(id) {
   }
   return list;
 }
+
+async function updateProductsInList(id, product) {
+  const listCollection = await getCollection("lists");
+  const objectId = new ObjectID.createFromHexString(id);
```
Comment (manuelpapa) [inline lib/models/list.js:35]: I would call it `ObjectId` to keep the naming consistent.

**Review by manuelpapa — COMMENTED** [review]
Great job, Jonas!

File: client/src/components/MatchingProductsDisplay.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    justify-content: space-around;
+    align-items: center;
+    background-color: var(--bg-color-dark-highlight);
+    box-shadow: 3px 3px 6px rgba(0, 0, 0, 0.16);
+
+    h4 {
+      color: var(--font-color-menu);
+      text-align: center;
+    }
+    > button {
```
Comment (manuelsanchez2) [inline client/src/components/MatchingProductsDisplay.js:51]: do you need the ` >` ?

**Review by manuelsanchez2 — APPROVED** [review]
looks really good! well done!

File: client/src/components/MatchingProductsDisplay.js (hunk 3 above)
Comment (LebkuchenCJ) [reply client/src/components/MatchingProductsDisplay.js:51]: Yes

File: lib/models/list.js (hunk 2 above)
Comment (LebkuchenCJ) [reply lib/models/list.js:35]: but here it's a new const which I created, therefore it will be written in camelCase, or?

**Pull request merged**